vendor/github.com/pachyderm/s2/error.go

```diff
@@ -32,6 +32,18 @@ func NewError(r *http.Request, httpStatus int, code string, message string) *Err
 	}
 }
 
+// NewFromGenericError takes in a generic error, and returns an s2 `Error`. If
+// the input error is not already an s2 `Error`, it is turned into an
+// `InternalError`.
+func NewFromGenericError(r *http.Request, err error) *Error {
```

Comment:
Not super familiar with context, but it doesn't seem this is used outsied the s2 package, will it be? Does it need to be exported?

Reply:
It doesn't; I'll tackle: https://github.com/pachyderm/s2/issues/20